src/frontend/src/helpers/Storage/WebStorage.ts

```diff
@@ -0,0 +1,588 @@
+import {SeashellWebsocket, WebsocketResult, WebsocketError} from "./WebsocketClient";
+import {Connection} from "../Login";
+import {LocalStorage} from "./LocalStorage";
+import {AbstractStorage,
+        Project, ProjectID, ProjectBrief,
+        File, FileID, FileBrief,
+        Settings, defaultSettings} from "./Interface";
+import {History,Test,SeashellFile,SeashellCompiler,SeashellRunner,SeashellTester,SeashellPID,Change} from "../types";
+export {Connection, WebsocketError, WebStorage, SeashellWebsocket}
+import * as R from "ramda";
+
+enum OfflineMode {Off, On, Forced}
+
+class Callback {
+  constructor(public type: string, public cb: (message?: any) => any, public now: boolean) { }
+}
+
+class WebStorage extends AbstractStorage {
+  private connection: Connection;
+  // private synced: boolean;
+  private connected: boolean;
+  private failed: boolean;
+  // private isSyncing: boolean;
```

Comment:
We need the isSyncing variable or something similar to provide a visual cue when syncing